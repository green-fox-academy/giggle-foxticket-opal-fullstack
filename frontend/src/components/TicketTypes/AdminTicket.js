import React from 'react';
import './AdminTicket.style.sass';
import GiveIcon from '../DynamicIcon/DynamicIcon';
import Button from '../Button/Button';

function AdminTicket({ title, description, iconName, buttonText }) {
  return (
    <div className="admin_container">
      <div className="admin_icon">
        <GiveIcon icon={iconName} />
      </div>
      <div className="admin_content">
        <h1 className="admin_title">{title}</h1>
        <p className="admin_description">{description}</p>
      </div>
      <div className="btns">
        <Button buttonStyle="btn--danger--solid">{buttonText}</Button>
        <Button buttonStyle="btn--danger--solid--btn">{buttonText}</Button>
      </div>
    </div>
  );
}
export default AdminTicket;
