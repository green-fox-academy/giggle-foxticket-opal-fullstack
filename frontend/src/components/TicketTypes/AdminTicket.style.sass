.container
  width: 50em
  height: 5.5em

  h1 
    margin-bottom: 0
  p 
    margin-top: 0
.icon
  display: flex
  justify-content: space-around
  align-items: center

  svg
    width: 40px
    height: 40px
.btns
    display: flex
    justify-content: space-around
    align-items: center 
    width: 200px
