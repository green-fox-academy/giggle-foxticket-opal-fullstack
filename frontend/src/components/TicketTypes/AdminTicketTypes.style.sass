.main-title
  font-size: 5em
  margin: 0
