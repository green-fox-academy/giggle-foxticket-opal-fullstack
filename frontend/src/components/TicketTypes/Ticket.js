import React from 'react';
import { FaCircle } from 'react-icons/fa';

function TicketType({ title, description }) {
  return (
    <div className="container">
      <FaCircle />
      <h1>{title}</h1> <p>{description}</p>
      <button className="edit"> EDIT </button>
      <button className="delete"> DELETE </button>
    </div>
  );
}
export default TicketType;
