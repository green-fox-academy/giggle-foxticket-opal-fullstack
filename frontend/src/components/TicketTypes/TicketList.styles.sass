.ticket-list-container
  display: grid
  grid-template-columns: repeat(3, 1fr)
  grid-auto-rows: minmax(150px, auto)
  grid-gap: 10px
  grid-template-areas: '. container .'
  margin: 0 auto


  .ticket-list
    grid-area: container
    display: inline-grid


    .btn--danger--solid
      transform: translateY(50%)

    .admin_content
      margin: 0
