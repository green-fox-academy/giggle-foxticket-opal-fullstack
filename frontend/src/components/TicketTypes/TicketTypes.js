import React from 'react';
import TicketType from './Ticket';

const fakeData = [
  {
    id: 1,
    title: 'Ticket-Types A',
    description: 'asdf,ffff,f,f,f,f,f,fffffdrdbdb kjbk ',
  },
  {
    id: 2,
    title: 'Ticket-Types B',
    description: 'asdf,ffff,f,f,f,f,f,fffffdrdbdbsefwa ',
  },
  {
    id: 3,
    title: 'Ticket-Types C',
    description: 'asdf,ffff,f,f,f,f,f,fffffdrdbdbfdsf ',
  },
  {
    id: 4,
    title: 'Ticket-Types D',
    description: 'asdf,ffff,f,f,f,f,f,fffffdrdbdbwtg ',
  },
];

const toRender = fakeData.map(i => {
  return <TicketType key={i.id} {...i} />;
});

function TicketTypes() {
  return (
    <div>
      {toRender}
    </div>
  );
}

export default TicketTypes;
