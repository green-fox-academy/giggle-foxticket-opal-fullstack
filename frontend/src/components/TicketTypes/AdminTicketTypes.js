import React from 'react';
import AdminTicket from './AdminTicket';
import './AdminTicketTypes.style.sass';

const fakeData = [
  {
    id: 1,
    title: 'Ticket-Types A',
    description: 'Lorem Ipsum has been the industry s standard dummy',
    iconName: 'FaBeer',
  },
  {
    id: 2,
    title: 'Ticket-Types B',
    description:'Since the 1500s,when an unknown printer took a galley',
    iconName: 'FaRegSmileWink',
  },
  {
    id: 3,
    title: 'Ticket-Types C',
    description:'Remaining  essentially unchanged. It was popularised',
    iconName: 'FaRegGem',
  },
  {
    id: 4,
    title: 'Ticket-Types D',
    description: 'Contrary to popular Lorem Ipsum is not  random text',
    iconName: 'FaUmbrella',
  },
];

function AdminTicketTypes() {
  return (
    <div>
      <h1 className="main-title">Ticket Types</h1>
      {fakeData.map(i => (
        <AdminTicket key={i.id} {...i} />
      ))}
    </div>
  );
}

export default AdminTicketTypes;
