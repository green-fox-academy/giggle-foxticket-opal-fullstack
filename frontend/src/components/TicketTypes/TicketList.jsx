import React from 'react';
import './TicketList.styles.sass';
import AdminTicket from './AdminTicket';

const ticketTypes = [
  {
    id: 1,
    title: 'Ticket-Types A',
    description: 'Lorem Ipsum has been the industry s standard dummy',
    iconName: 'FaBeer',
  },
  {
    id: 2,
    title: 'Ticket-Types B',
    description: 'Since the 1500s,when an unknown printer took a galley',
    iconName: 'FaRegSmileWink',
  },
  {
    id: 3,
    title: 'Ticket-Types C',
    description: 'Remaining  essentially unchanged. It was popularised',
    iconName: 'FaRegGem',
  },
  {
    id: 4,
    title: 'Ticket-Types D',
    description: 'Contrary to popular Lorem Ipsum is not  random text',
    iconName: 'FaUmbrella',
  },
];

const TicketList = () => {
  return (
    <div className="ticket-list-container">
      <div className="ticket-list">
        <h1 className="main-title">Ticket Types</h1>
        {ticketTypes.map(i => (
          <AdminTicket key={i.id} {...i} />
        ))}
      </div>
    </div>
  );
};

export default TicketList;
