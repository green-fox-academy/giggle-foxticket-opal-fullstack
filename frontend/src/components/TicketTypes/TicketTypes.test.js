import React from 'react';
import TicketTypes from './TicketTypes';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(<TicketTypes />).toJSON();
  expect(tree).toMatchSnapshot();
});
