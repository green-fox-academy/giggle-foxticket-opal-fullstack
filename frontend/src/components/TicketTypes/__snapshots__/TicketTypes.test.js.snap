// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h1
    style={
      Object {
        "fontSize": "5em",
      }
    }
  >
    Ticket Types
  </h1>
  <div
    className="container"
  >
    <div
      className="icon"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": "lightgreen",
            "fontSize": "3em",
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
        />
      </svg>
    </div>
    <div
      className="content"
    >
      <h1>
        Ticket-Types A
      </h1>
      <p>
        asdf,ffff,f,f,f,f,f,fffffdrdbdb kjbk 
      </p>
    </div>
    <div
      className="btns"
    >
      <button
        className="btn btn--danger--solid btn--medium"
      >
        Edit
      </button>
      <button
        className="btn btn--danger--solid--btn btn--medium"
      >
        Delete
      </button>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="icon"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": "lightgreen",
            "fontSize": "3em",
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
        />
      </svg>
    </div>
    <div
      className="content"
    >
      <h1>
        Ticket-Types B
      </h1>
      <p>
        asdf,ffff,f,f,f,f,f,fffffdrdbdbsefwa 
      </p>
    </div>
    <div
      className="btns"
    >
      <button
        className="btn btn--danger--solid btn--medium"
      >
        Edit
      </button>
      <button
        className="btn btn--danger--solid--btn btn--medium"
      >
        Delete
      </button>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="icon"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": "lightgreen",
            "fontSize": "3em",
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
        />
      </svg>
    </div>
    <div
      className="content"
    >
      <h1>
        Ticket-Types C
      </h1>
      <p>
        asdf,ffff,f,f,f,f,f,fffffdrdbdbfdsf 
      </p>
    </div>
    <div
      className="btns"
    >
      <button
        className="btn btn--danger--solid btn--medium"
      >
        Edit
      </button>
      <button
        className="btn btn--danger--solid--btn btn--medium"
      >
        Delete
      </button>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="icon"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": "lightgreen",
            "fontSize": "3em",
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
        />
      </svg>
    </div>
    <div
      className="content"
    >
      <h1>
        Ticket-Types D
      </h1>
      <p>
        asdf,ffff,f,f,f,f,f,fffffdrdbdbwtg 
      </p>
    </div>
    <div
      className="btns"
    >
      <button
        className="btn btn--danger--solid btn--medium"
      >
        Edit
      </button>
      <button
        className="btn btn--danger--solid--btn btn--medium"
      >
        Delete
      </button>
    </div>
  </div>
</div>
`;
